src/Models/LagrangianParticleTracking/lagrangian_particle_advection.jl

```diff
@@ -48,27 +48,26 @@ bouncing the particle off the immersed boundary with a coefficient or `restituti
     X = flattened_node((x, y, z), ibg)
 
     # Determine current particle cell
-    fi, fj, fk = fractional_indices(X, ibg.underlying_grid, (c, c, c))
+    fi, fj, fk = fractional_indices(X, ibg.underlying_grid, c, c, c)
     i, j, k = truncate_fractional_indices(fi, fj, fk)
 
     if immersed_cell(i, j, k, ibg)
         # Determine whether particle was _previously_ in a non-immersed cell
         i⁻, j⁻, k⁻ = previous_particle_indices
 
-        if !immersed_cell(i⁻, j⁻, k⁻, ibg)
-            # Left-right bounds of the previous, non-immersed cell
-            xᴿ, yᴿ, zᴿ = node(i⁻+1, j⁻+1, k⁻+1, ibg, f, f, f)
-            xᴸ, yᴸ, zᴸ = node(i⁻,   j⁻,   k⁻,   ibg, f, f, f)
+        # Left-right bounds of the previous, non-immersed cell
+        xᴿ, yᴿ, zᴿ = node(i⁻ + 1, j⁻ + 1, k⁻ + 1, ibg, f, f, f)
+        xᴸ, yᴸ, zᴸ = node(i⁻,     j⁻,     k⁻,     ibg, f, f, f)
 
-            Cʳ = restitution
-            x⁺ = enforce_boundary_conditions(Bounded(), x, xᴸ, xᴿ, Cʳ)
-            y⁺ = enforce_boundary_conditions(Bounded(), y, yᴸ, yᴿ, Cʳ)
-            z⁺ = enforce_boundary_conditions(Bounded(), z, zᴸ, zᴿ, Cʳ)
-
-        end
+        Cʳ = restitution
+        x⁺ = enforce_boundary_conditions(Bounded(), x, xᴸ, xᴿ, Cʳ)
+        y⁺ = enforce_boundary_conditions(Bounded(), y, yᴸ, yᴿ, Cʳ)
+        z⁺ = enforce_boundary_conditions(Bounded(), z, zᴸ, zᴿ, Cʳ)
+    else
+        x⁺, y⁺, z⁺ = x, y, z
```

Comment:
This part of the algorithm was failing so I had to change it. However, I'm not familiar with the lagrangian advection algorithm so I'm sure if this is the right way to do things. Can someone double-check this part? From what I can tell, @Jamie-Hilditch and @jagoosw were the main authors of this algorithm, so maybe we should hear from them also.

Reply:
It might be cleaner todo

```
if immersed_cell
    ...
    return ...
else
    return x, y, z
end
```

But otherwise I agree this looks like it is needed.

Reply:
you can also use an `ifelse` syntax to avoid branching:
```julia
immersed = immersed_cell(i, j, k, ibg)  
x⁺ = ifelse(immersed, enforce_boundary_conditions(Bounded(), x, xᴸ, xᴿ, Cʳ), x)
y⁺ = ifelse(immersed, enforce_boundary_conditions(Bounded(), y, yᴸ, yᴿ, Cʳ), y)
z⁺ = ifelse(immersed, enforce_boundary_conditions(Bounded(), z, zᴸ, zᴿ, Cʳ), z)
```